Update Test documentation about test database behaviours (#9292)
silverstripe/silverstripe-framework: docs/en/02_Developer_Guides/06_Testing/00_Unit_Testing.md

## Test Databases and Fixtures

> SilverStripe tests create their own database when the test starts and fixture files are specified. New `ss_tmp` databases are created using the same 

Reviewer: ```suggestion
SilverStripe tests create their own database when the test starts and fixture files are specified, or `protected $usesDatabase = true` is set on the test class. New `ss_tmp` databases are created using the same 
```

Author: Native English speaker FTW

</div>

<div class="notice" markdown="1">

> The test database is rebuilt every time one of the test methods is run and cleaned up afterwards. If the test is interrupted the database is not cleaned up, over time, you may have several hundred test 

Reviewer: ```suggestion
The test database is rebuilt every time one of the test methods is run and is removed afterwards. If the test is interrupted, the database will not be removed. Over time, you may have several hundred test 
```
